controller/controller.go

```diff
@@ -208,10 +211,14 @@ func (c *Controller) Reconcile(req reconcile.Request) (reconcile.Result, error)
 
 	// Add common keys to the logger context.
 	{
-		loop := atomic.AddInt64(&c.loop, 1)
-
 		ctx = setLoggerCtxValue(ctx, loggerKeyController, c.name)
-		ctx = setLoggerCtxValue(ctx, loggerKeyLoop, strconv.FormatInt(loop, 10))
+		ctx = setLoggerCtxValue(ctx, loggerKeyLoop, strconv.FormatInt(atomic.AddInt64(&c.loop, 1), 10))
+	}
+
+	{
+		ctx = cachekeycontext.NewContext(ctx)
+		ctx = finalizerskeptcontext.NewContext(ctx)
+		ctx = updateallowedcontext.NewContext(ctx)
```

Comment:
Motivation for the interface change can be seen here. The call is easier and more straight forward now. There was never an apparent reason to provide the channel we only use internally.